crates/bevy_ui/src/update.rs

```diff
@@ -37,35 +37,46 @@ fn update_clipping(
     entity: Entity,
     clip: Option<Rect>,
 ) {
-    let (node, global_transform, style, calculated_clip) = node_query.get_mut(entity).unwrap();
-    // Update this node's CalculatedClip component
-    match (clip, calculated_clip) {
+    let (node, global_transform, style, old_clip) = node_query.get_mut(entity).unwrap();
+
+    // Update current node's CalculatedClip component
+    match (old_clip, clip) {
```

Comment:
`(old_clip, clip)` is an improvement but I think the semantics would be even clearer if you renamed `clip` as well so it's 

```suggestion
    match (old_clip, new_clip) {
```